app/models/operatingsystems/nxos.rb

```diff
@@ -42,7 +42,7 @@ def downcase_release_name
 
   # generate a Cisco release number using release_name as an auxiliary field
   def release
-    "#{major}#{('.' + minor.to_s) if minor.present?}#{('.' + release_name) if release_name.present?}"
+    "#{major}#{('.' + minor.to_s) if minor.present?}#{('.' + release.to_s) if release.present?}#{('.' + release_name) if release_name.present?}"
```

Comment:
We need to rename the method as well, otherwise we will stuck in infinite loop (we call `release` in method `release`)